streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -576,6 +599,182 @@ public void shouldNotBlockInCloseForZeroDuration() throws InterruptedException {
         }
     }
 
+    @Test
+    public void statelessTopologyShouldNotCreateStateDirectory() throws Exception {
```

Comment:
Isn't this case covered by `statelessPAPITopologyShouldNotCreateStateDirectory` ? DSL compiles down into a Topology anyway.

Similar below.